includes/polling/applications/bird2.inc.php

```diff
@@ -0,0 +1,192 @@
+            $IpVersion = 6;
+        }
+
+        if (strpos($protocolBody, 'hannel ipv' . $IpVersion) === 0) {
+
+            foreach (explode("\n", 'C' . $protocolBody) as $protocolBodyLine) {
+                if (strpos($protocolBodyLine, ':') !== false) {
+                    $lineParts = explode(':', $protocolBodyLine, 2);
+                    $protocolData[str_replace(' ', '_', strtolower(trim($lineParts[0])))] = trim($lineParts[1]);
+                }
+            }
+
+            // Fix up the ROUTES
+            if (isset($protocolData['routes'])) {
+                $routeParts = explode(', ', $protocolData['routes']);
+                unset($protocolData['routes']);
+                foreach ($routeParts as $routePart) {
+                    $routeDetail = explode(' ', $routePart);
+                    $protocolData['routes'][$routeDetail[1]] = $routeDetail[0];
+                }
+            }
+
+            // Set the route updates
+            unset($protocolData['route_change_stats']);
+            foreach (['import_updates', 'import_withdraws', 'export_updates', 'export_withdraws'] as $key) {
+                if (!isset($protocolData[$key])) {
+                    continue;
+                }
+
+                $routeChange_parts = preg_split("/\s+/", trim($protocolData[$key]));
+
+                unset($protocolData[$key]);
+                $protocolData['route_change_stats'][$key] = [
+                    'received' => $routeChange_parts[0],
+                    'rejected' => $routeChange_parts[1],
+                    'filtered' => $routeChange_parts[2],
+                    'ignored' => $routeChange_parts[3],
+                    'accepted' => $routeChange_parts[4],
+                ];
+            }
+        }
+    }
+
+    $protocolsData[] = $protocolData;
+}
+
+// ---
+$deviceObj = Device::findOrFail($device['device_id']);
+
+if (empty($protocolsData)) {
+    echo PHP_EOL . $name . ': No BGP Peers found' . PHP_EOL;
+    $deviceObj->bgpLocalAs = 'NULL';
+    $deviceObj->save();
+    return;
+}
+
+// Do bgpLocalAs Update
+// Get the most common localAS (in theory there *should* be only one, but not always
+$localAsns = array_count_values(array_column($protocolsData, 'local_as'));
+arsort($localAsns);
+$bgpLocalAs = array_keys($localAsns)[0];
+
+$deviceObj->bgpLocalAs = $bgpLocalAs;
+$deviceObj->save();
+
+// Going through all BGP Peers
+$bgpPeerIds = [];
+
+foreach ($protocolsData as $protocol) {
+    $bgpPeer = BgpPeer::firstOrNew([
+        'device_id' => $device['device_id'],
+        'bgpPeerRemoteAs' => $protocol['neighbor_as'],
+        'bgpLocalAddr' => $protocol['source_address'] ?: '0.0.0.0',
+        'bgpPeerRemoteAddr' => $protocol['neighbor_address'],
+    ]);
+
+    $bgpPeer->device_id = $device['device_id'];
+    $bgpPeer->astext = get_astext($protocol['neighbor_as']);
+    $bgpPeer->bgpPeerIdentifier = $protocol['neighbor_id'] ?: '0.0.0.0';
```

Comment:
Commentor indicated there is an issue with neighbor_id, does it exist?

Reply:
I think there is some confusions between `neighbour address` and `neighbour id`

- neighbour address = This is the actual address used to establish the BGP session (most people care about this)
-  neighbour id = this is the `router-id` which is set as an arbitrary ipv4 address, from my understabnding, no one uses or cares about this, its a simple `id` in the form of an ipv4 address (even for an ipv6 session this router-id is used as a ipv4 address. 

Here I tried to remain consistent with the naming of the variables, `bgpPeerIdentifier` as the `ID` while `bgpPeerRemoteAddr` is the actual address to establish the BGP session. 

This is another reason why its less confusing to display the `remote address` and NOT `id` here: https://github.com/librenms/librenms/blob/master/includes/html/pages/routing/bgp.inc.php#L283C15-L283C15

Where it uses `peer_ident` which is **NOT** the peer's actual IPv4/IPv6 address but the `router-id` 


tl;dr with the current commit I tried to stay as consistant as possible with the current way librenms displays the peer address (even though I believe that its done wrong in `includes/html/pages/routing/bgp.inc.php`

Reply:
@murrant On second thought I figured it will be actually to fix the bug on the display of the LibreNMS - see the additional commit here: https://github.com/librenms/librenms/pull/14466/commits/2d3210eee99812cc1e2187f198c4e0311cf794bd